google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/SampleRowKeys.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigtable.data.v2.models;
+
+import com.google.api.core.InternalApi;
+import com.google.bigtable.v2.SampleRowKeysRequest;
+import com.google.cloud.bigtable.data.v2.internal.NameUtil;
+import com.google.cloud.bigtable.data.v2.internal.RequestContext;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import java.io.Serializable;
+import javax.annotation.Nonnull;
+
+/** Wraps a {@link com.google.bigtable.v2.SampleRowKeysRequest}. */
+public final class SampleRowKeys implements Serializable {
+  private final TargetId targetId;
+
+  private SampleRowKeys(@Nonnull TargetId targetId) {
+    Preconditions.checkNotNull(targetId, "target id can't be null.");
+
+    this.targetId = targetId;
+  }
+
+  /** Creates a new instance of the sample row keys builder for the given target with targetId */
+  public static SampleRowKeys create(@Nonnull TargetId targetId) {
+    Preconditions.checkNotNull(targetId, "target id can't be null.");
```

Comment:
We don't need to checkNotNull here because the ctor is already checking it. Also no need for Nonnull annotation